homeassistant/components/radiotherm/config_flow.py

```diff
@@ -0,0 +1,171 @@
+"""Config flow for Radio Thermostat integration."""
+from __future__ import annotations
+
+import logging
+from socket import timeout
+from typing import Any
+
+from radiotherm.validate import RadiothermTstatError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import dhcp
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import CONF_HOLD_TEMP, DOMAIN
+from .data import RadioThermInitData, async_get_init_data
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class CannotConnect(HomeAssistantError):
+    """Error to indicate we cannot connect."""
+
+
+async def validate_connection(hass: HomeAssistant, host: str) -> RadioThermInitData:
+    """Validate the connection."""
+    try:
+        return await async_get_init_data(hass, host)
+    except (timeout, RadiothermTstatError) as ex:
+        raise CannotConnect(f"Failed to connect to {host}: {ex}") from ex
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Radio Thermostat."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize ConfigFlow."""
+        self.discovered_ip: str | None = None
+        self.discovered_init_data: RadioThermInitData | None = None
+
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
+        """Discover via DHCP."""
+        self._async_abort_entries_match({CONF_HOST: discovery_info.ip})
+        try:
+            init_data = await validate_connection(self.hass, discovery_info.ip)
+        except CannotConnect:
+            return self.async_abort(reason="cannot_connect")
+        await self.async_set_unique_id(init_data.mac)
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: discovery_info.ip, CONF_HOLD_TEMP: False},
```

Comment:
Why override `CONF_HOLD_TEMP` ?

Reply:
Good catch. That's a refactoring error.

54e57689c0a9db2d486a05a2b5bd86f64f8ec4f8